Tools/clinic/clinic.py

```diff
@@ -3005,7 +3005,7 @@ def parse_arg(self, argname, argnum):
                     goto exit;
                 }}}}
                 {paramname} = PyLong_AsLongLong({argname});
-                if ({paramname} == (PY_LONG_LONG)-1 && PyErr_Occurred()) {{{{
+                if ({paramname} == -1 && PyErr_Occurred()) {{{{
```

Comment:
I decided to remove this cast as it's unneeded.

Reply:
@serhiy-storchaka you committed this code in 32d96a2b5bc3136d45a66adbdb45fac351b520ce. Does it make sense to you to remove this cast?

Reply:
It was copied from `Python/getargs.c`.

Reply:
I would prefer to keep the explicit cast, since long long is an uncommon type and I prefer to avoid bad surprises on some platforms.

Reply:
@vstinner I don't really see a problem here, because implicit conversion of integers of the same signedness is well defined and always lossless. If it's so important it can be rewritten as: 
```c
if ({paramname} == -1LL && PyErr_Occurred()) {{{{
``` 
In that case there is no casts at all.

Reply:
Ok.

Reply:
> I don't really see a problem here, because implicit conversion of integers of the same signedness is well defined and always lossless

@vstinner @serhiy-storchaka So is this okay universally across the C-API or is it different in other areas? Of course it wouldn't be worth a PR on it's own since it's more of a conventional/styling decision, I just want to know in case I see a similar implicit integer conversion (signed -> signed or unsigned -> unsigned) without the cast in another PR or if the issue comes up again.

Reply:
I have no idea. The C language remains partially a mystery to me.

Reply:
@vstinner 
> I have no idea. The C language remains partially a mystery to me.

If the C language is a mystery even to you, what hope do the rest of us mere mortals have? Haha.